sycl/test-e2e/AddressSanitizer/bad-free/bad-free-minus1.cpp

```diff
@@ -0,0 +1,29 @@
+// REQUIRES: linux, cpu
+// RUN: %{build} %device_sanitizer_flags -O0 -g -o %t
+// RUN: env SYCL_PREFER_UR=1 UR_ENABLE_LAYERS=UR_LAYER_ASAN %{run} not %t 2>&1 | FileCheck --check-prefixes CHECK,CHECK-DEVICE %s
```

Comment:
Can we move `env SYCL_PREFER_UR=1 UR_ENABLE_LAYERS=UR_LAYER_ASAN` into a substitution? I think local lit config should allow that.

Reply:
Moved device sanitizer specific substitution into "sycl/test-e2e/AddressSanitizer/lit.local.cfg"